torch/csrc/jit/script/init.cpp

```diff
@@ -620,6 +628,30 @@ void initJitScriptBindings(PyObject* module) {
         cu->define(methodDefs, rcbs, simpleSelf(classType));
       });
 
+  m.def(
+      "_jit_script_named_tuple_compile",
+      [](const std::string& qualifiedName,
+         const std::vector<std::string>& fields,
```

Comment:
we should be passing a `TreeView` here, similar to how the other compile functions work. I think the python side should be responsible for lowering something like `namedtuple("foo", ["bar", str])` into the right `TreeView` (probably a classdef?).

This will ensure that when we eventually merge the two concepts they aren't too out of sync